nexus/src/app/instance.rs

```diff
@@ -2186,18 +2166,114 @@ pub(crate) async fn notify_instance_updated(
     }
 }
 
+/// Determines the disposition of a request to start an instance given its state
+/// (and its current VMM's state, if it has one) in the database.
+fn instance_start_allowed(
+    log: &slog::Logger,
+    instance: &DbInstance,
+    vmm: &Option<DbVmm>,
```

Comment:
nitpicky: i kinda wonder if this ought to take `&InstanceAndActiveVmm` instead of `&Instance, &Vmm`, because that kind of enforces the invariant that you should call it with an instance and _its active VMM_ rather than any random VMM?

OTTOH, that probably makes the tests annoyinger.

Reply:
I was worried about this when I started writing the tests, but it actually turns out not to be so bad, and it gets rid of some of the `unwrap()` nonsense you mentioned in the previous comment. Will push a fix for this shortly.